src/main/java/org/spongepowered/api/map/font/MapFont.java

```diff
@@ -0,0 +1,129 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.font;
+
+import com.google.common.collect.ImmutableMap;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents a font on a map that can be used for drawing characters.
+ */
+public final class MapFont {
+
+    private final ImmutableMap<Character, CharacterSprite> characters;
+
+    public MapFont(Map<Character, CharacterSprite> characters, int spaceSize) {
+        ImmutableMap.Builder<Character, CharacterSprite> builder =
+                new ImmutableMap.Builder<Character, CharacterSprite>();
+
+        if (!characters.containsKey(' ')) {
+            builder.put(' ', new CharacterSprite(spaceSize, 8, new boolean[16]));
+        }
+
+        builder.putAll(characters);
+
+        this.characters = builder.build();
+    }
+
+    /**
+     * Returns the {@link CharacterSprite} data for the specified character.
+     *
+     * @param ch The character to retrieve a sprite for
+     * @return The sprite version of the character
+     */
+    public Optional<CharacterSprite> getChar(char ch) {
+        return Optional.ofNullable(characters.get(ch));
+    }
+
+    /**
+     * Returns the pixel width of a string of text accounting for the
+     * {@link CharacterSprite} sizes and a spacing between characters.
+     *
+     * @param text The string of text to analyze the width of
+     * @return The width of the text in pixels
+     * @throws IllegalArgumentException If the text provided contains a
+     *         character not provided in the character sprites
+     */
+    public int getWidth(String text) {
+        checkNotNull(text, "text");
+
+        if (text.length() == 0) {
+            return 0;
+        }
+
+        int width = 0;
+        // Avoids copying the inner character array by indexing
+        for (int i = 0; i < text.length(); i++) {
+            char ch = text.charAt(i);
+            CharacterSprite value = characters.get(ch);
+            if (value == null) {
+                throw new IllegalArgumentException("Character '" + ch +
```

Comment:
`checkArgument(value != null, "Character '" + ch + "' was not provided as a CharacterSprite.");`